### PR TITLE
CIRC-782: Fixed Due Date Range is not recognized if End Date = Current Date

### PR DESCRIPTION
**Pull request by Illia-Dolzhenko**
## Purpose

- Add time value to a due date before check

Resolves: https://issues.folio.org/browse/CIRC-782

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/policy/FixedDueDateSchedules.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public static FixedDueDateSchedules from(JsonObject representation) {
   private Predicate<? super JsonObject> isWithin(DateTime date) {
     return schedule -> {
       DateTime from = DateTime.parse(schedule.getString("from"));
-      DateTime to = DateTime.parse(schedule.getString("to"));
+      DateTime to = DateTime.parse(schedule.getString("to")).withTime(23,59,59,0);
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/policy/FixedDueDateSchedules.java:51]:
```suggestion
      DateTime to = DateTime.parse(schedule.getString("to")).withTime(23, 59, 59, 0);
```

**Review by marcjohnson-kint — COMMENTED** [review]
Given this change is overriding the time information provided by the client when fixed due date schedules are set up, please add a test that this works as expected in a non-UTC timezone.

**Review by marcjohnson-kint — APPROVED** [review]

File: src/main/java/org/folio/circulation/domain/policy/FixedDueDateSchedules.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public static FixedDueDateSchedules from(JsonObject representation) {
   private Predicate<? super JsonObject> isWithin(DateTime date) {
     return schedule -> {
       DateTime from = DateTime.parse(schedule.getString("from"));
-      DateTime to = DateTime.parse(schedule.getString("to"));
+      DateTime to = DateTime.parse(schedule.getString("to")).withTime(23, 59, 59, 0);
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/domain/policy/FixedDueDateSchedules.java:51]: I think it worth moving this logic to `DateTimeUtil` utility.

File: src/test/java/org/folio/circulation/domain/policy/FixedLoanPolicyCheckOutDueDateCalculationTests.java [hunk 3]
```diff
@@ -39,6 +39,51 @@ public void shouldUseOnlyScheduleAvailableWhenLoanDateFits() {
...
+
+    DateTime loanDate = new DateTime(2020, 11, 2, 12, 30, 30, DateTimeZone.UTC);
+
+    Loan loan = loanFor(loanDate);
+
+    final Result<DateTime> calculationResult = loanPolicy
+      .calculateInitialDueDate(loan, null);
+
+    assertThat(calculationResult.value(), is(new DateTime(2020, 11, 2, 0, 0, 0,
+      DateTimeZone.UTC)));
```
Comment (bohdan-suprun) [inline src/test/java/org/folio/circulation/domain/policy/FixedLoanPolicyCheckOutDueDateCalculationTests.java:61]:
```suggestion
    final var expectedInitialDueDate = new DateTime(2020, 11, 2, 0, 0, 0,
      DateTimeZone.UTC));
      
    assertThat(calculationResult.succeeded(), is(true));
    assertThat(calculationResult.value(), is();
```

File: src/main/java/org/folio/circulation/domain/policy/FixedDueDateSchedules.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/circulation/domain/policy/FixedDueDateSchedules.java:51]: There is already `org.folio.circulation.support.utils.DateTimeUtil#atEndOfTheDay` method but for java date time class. We can add similar but for joda date time.

**Review by bohdan-suprun — APPROVED** [review]

**Pull request merged**